From c909ef300d19f181fa2cf6f70daad17c5d61fb96 Mon Sep 17 00:00:00 2001
From: Lei Zhang <thestig@chromium.org>
Date: Tue, 23 Jul 2019 02:02:05 +0000
Subject: [PATCH] Do IWYU in a few header files.

https://crrev.com/675319 added more std::unique_ptr usage, but did not
always #include <memory>. As a result, the CL only builds due to luck.
When one sets use_custom_cxx=false, they no longer build, as seen in
this thread:
https://groups.google.com/a/chromium.org/d/topic/chromium-packagers/UyJsVJ5QqWo/discussion

Change-Id: Ib9812b3210dac60176128b22c3881f7f33d785f8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1713037
Reviewed-by: Scott Violet <sky@chromium.org>
Commit-Queue: Lei Zhang <thestig@chromium.org>
Cr-Commit-Position: refs/heads/master@{#679819}
---

diff --git a/chrome/browser/ui/bookmarks/bookmark_drag_drop.h b/chrome/browser/ui/bookmarks/bookmark_drag_drop.h
index fcf9b05..28c92a7 100644
--- a/chrome/browser/ui/bookmarks/bookmark_drag_drop.h
+++ b/chrome/browser/ui/bookmarks/bookmark_drag_drop.h
@@ -5,6 +5,7 @@
 #ifndef CHROME_BROWSER_UI_BOOKMARKS_BOOKMARK_DRAG_DROP_H_
 #define CHROME_BROWSER_UI_BOOKMARKS_BOOKMARK_DRAG_DROP_H_
 
+#include <memory>
 #include <vector>
 
 #include "build/build_config.h"
diff --git a/components/exo/wm_helper.h b/components/exo/wm_helper.h
index 83f0b24..9c9de973 100644
--- a/components/exo/wm_helper.h
+++ b/components/exo/wm_helper.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_EXO_WM_HELPER_H_
 #define COMPONENTS_EXO_WM_HELPER_H_
 
+#include <memory>
 #include <vector>
 
 #include "base/macros.h"
diff --git a/ui/aura/client/drag_drop_client.h b/ui/aura/client/drag_drop_client.h
index 8de8940..3d08256 100644
--- a/ui/aura/client/drag_drop_client.h
+++ b/ui/aura/client/drag_drop_client.h
@@ -5,6 +5,8 @@
 #ifndef UI_AURA_CLIENT_DRAG_DROP_CLIENT_H_
 #define UI_AURA_CLIENT_DRAG_DROP_CLIENT_H_
 
+#include <memory>
+
 #include "ui/aura/aura_export.h"
 #include "ui/base/dragdrop/drag_drop_types.h"
 #include "ui/gfx/native_widget_types.h"
diff --git a/ui/aura/client/drag_drop_delegate.h b/ui/aura/client/drag_drop_delegate.h
index 304b1a6..82a2dc6 100644
--- a/ui/aura/client/drag_drop_delegate.h
+++ b/ui/aura/client/drag_drop_delegate.h
@@ -5,6 +5,8 @@
 #ifndef UI_AURA_CLIENT_DRAG_DROP_DELEGATE_H_
 #define UI_AURA_CLIENT_DRAG_DROP_DELEGATE_H_
 
+#include <memory>
+
 #include "ui/aura/aura_export.h"
 #include "ui/aura/window.h"
 #include "ui/base/dragdrop/os_exchange_data.h"
diff --git a/ui/views/drag_utils.h b/ui/views/drag_utils.h
index d8c1be0..1604d9e 100644
--- a/ui/views/drag_utils.h
+++ b/ui/views/drag_utils.h
@@ -5,6 +5,8 @@
 #ifndef UI_VIEWS_DRAG_UTILS_H_
 #define UI_VIEWS_DRAG_UTILS_H_
 
+#include <memory>
+
 #include "ui/base/dragdrop/drag_drop_types.h"
 #include "ui/base/dragdrop/os_exchange_data.h"
 #include "ui/gfx/native_widget_types.h"
